File: src/main/ui/src/components/DeviceService.js
import axios from "axios";

const DEVICE_API_BASE_URL = "http://localhost:8080/api/devices"

class DeviceService {

    getDevices() {
        return axios.get(DEVICE_API_BASE_URL);
    }

    createDevice(device) {
        return axios.post(DEVICE_API_BASE_URL, device);
    }

    getDeviceById(deviceId) {
        return axios.get(DEVICE_API_BASE_URL + "/" + deviceId);
    }

    updateDevice(device, deviceId) {
        //deviceId == 1 => http://localhost:8080/api/devices/1
        return axios.put(DEVICE_API_BASE_URL + "/" + deviceId, device);
    }

    deleteDevice(deviceId) {
        return axios.delete(DEVICE_API_BASE_URL + "/" + deviceId);
    }
}

export default new DeviceService()

File: src/main/java/com/example/smarthouse_v3/service/DeviceService.java
package com.example.smarthouse_v3.service;

import com.example.smarthouse_v3.domain.Device;
import org.springframework.http.ResponseEntity;

import java.util.Map;

public interface DeviceService {
    Iterable<Device> getAllDevices();

    Device createDevice(Device device);

    ResponseEntity<Device> getDeviceById(Long id);

    ResponseEntity<Device> updateDevice(Long id, Device extDevice);

    ResponseEntity<Map<String, Boolean>> deleteDevice(Long id);

    ResponseEntity<Device> updateErrors(Long id, Device extDevice);
}
